tempel.el

```diff
@@ -348,12 +363,30 @@ PROMPT is the optional prompt/default value."
 
 (defun tempel--save ()
   "Save template file buffer."
-  (when-let (buf (get-file-buffer tempel-file))
-    (with-current-buffer buf
-      (when (and (buffer-modified-p) (y-or-n-p (format "Save file %s? " tempel-file)))
-        (save-buffer buf)))))
-
-(defun tempel--file-read (file)
+  (let (modified)
+    (dolist (file (tempel--expand-path tempel-path))
+      (when-let (buffer (find-buffer-visiting file))
+    	(when (buffer-modified-p buffer)
+	  (push buffer modified))))
+    (cond ((not modified))
+	  ((not (cdr modified))
+	   (y-or-n-p "Save template file %s? " (car modified)))
+	  (t
+	   (y-or-n-p "Save modified template files? ")))
+    (mapc #'save-buffer modified)))
```

Comment:
The cond is not correct. `(mapc #'save-buffer modified)` will be executed in any case. Did you test your changes?

Reply:
I thought that if `modified` is nil `mapc` won't do anything. That's why I left it outside of the `cond` case. Is this not the case? Running `(mapc #'save-buffer nil)` does nothing for me.

Reply:
In other words, if modified is not empty `(mapc #'save-buffer modified)` is appropriate. But in any case of `modified` is nil nothing happens, which is the correct behavior, right?

Reply:
I am not talking about the nil case, but about the case where you ask for confirmation.

Reply:
> In other words, if modified is not empty (mapc #'save-buffer modified) is appropriate. But in any case of modified is nil nothing happens, which is the correct behavior, right?

No, if modified is non-empty, we ask for confirmation and save only when the user confirms.

Reply:
Ok. Got it.